go/ct/spc/specification.go

```diff
@@ -2362,3 +2397,11 @@ func sumWithOverflow(values ...vm.Gas) (vm.Gas, bool) {
 	}
 	return res, false
 }
+
+func wordCountWithOverflow(size uint64) (vm.Gas, bool) {
+	offsetted := size + 31
+	if offsetted < size {
+		return 0, true
+	}
+	return vm.Gas(offsetted / 32), false
+}
```

Comment:
There is a subtle semantic hidden in this code -- what to do with a `size ∈ [MaxUint64-30,MaxUint64]`.

For those values, you could compute a valid result by returning `MaxUint64/32`. However, this implementation chooses to signal an overflow. Is such an overflow actually an error? After all, it only happens for an intermediate value.

Maybe a cleaner implementation may be:
```suggestion
func getNumWords(size uint64) uint64 {
        if size > math.MaxUint64-31 {
		return math.MaxUint64/32 + 1
	}
	return (size + 31) / 32
}
```

Reply:
I looked it up in go-ethereum, and they actually define it like this: [here](https://github.com/Fantom-foundation/go-ethereum-sonic/blob/2400937cc3b1c370ea8735d7af86eac621e4eb70/core/state_transition.go#L120-L127)

`evmone` ignores the overflows: [here](https://github.com/ethereum/evmone/blob/705409fdcd06e3ffbcfc8955a3ad5acfd7e360f2/lib/evmone/instructions.hpp#L61-L64)